java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -2001,6 +2003,58 @@ public CompletableFuture<Object[][]> xpending(
                 XPending, args, response -> castArray(handleArrayResponse(response), Object[].class));
     }
 
+    @Override
+    public CompletableFuture<Map<String, String[]>> xclaim(
+            @NonNull String key,
+            @NonNull String group,
+            @NonNull String consumer,
+            long minIdleTime,
+            @NonNull String[] ids) {
+        String[] args =
+                concatenateArrays(new String[] {key, group, consumer, Long.toString(minIdleTime)}, ids);
+        return commandManager.submitNewCommand(XClaim, args, this::handleMapResponse);
+    }
+
+    @Override
+    public CompletableFuture<Map<String, String[]>> xclaim(
+            @NonNull String key,
+            @NonNull String group,
+            @NonNull String consumer,
+            long minIdleTime,
+            @NonNull String[] ids,
+            @NonNull StreamClaimOptions options) {
+        String[] args =
+                concatenateArrays(new String[] {key, group, consumer, Long.toString(minIdleTime)}, ids);
+        args = concatenateArrays(args, options.toArgs(false));
+        return commandManager.submitNewCommand(XClaim, args, this::handleMapResponse);
+    }
+
+    @Override
+    public CompletableFuture<String[]> xclaimJustId(
+            @NonNull String key,
+            @NonNull String group,
+            @NonNull String consumer,
+            long minIdleTime,
+            @NonNull String[] ids) {
+        return xclaimJustId(
+                key, group, consumer, minIdleTime, ids, StreamClaimOptions.builder().build());
+    }
+
+    @Override
+    public CompletableFuture<String[]> xclaimJustId(
+            @NonNull String key,
+            @NonNull String group,
+            @NonNull String consumer,
+            long minIdleTime,
+            @NonNull String[] ids,
+            @NonNull StreamClaimOptions options) {
+        String[] args =
+                concatenateArrays(new String[] {key, group, consumer, Long.toString(minIdleTime)}, ids);
+        args = concatenateArrays(args, options.toArgs(true));
```

Comment:
if `justid` can be the last argument, do
```suggestion
        args = concatenateArrays(args, options.toArgs(), JUSTID);
```